loader/loader_test.go

```diff
@@ -2126,6 +2126,16 @@ services:
 	assert.ErrorContains(t, err, "services.test.build.ssh must be a mapping")
 }
 
+func TestLoadLegacyBoolean(t *testing.T) {
+	_, err := loadYAML(`
```

Comment:
@glours 

Since the goal is to support legacy booleans, `actual` checking should not be removed in the test code.

Reply:
I agree that you don't want to support legacy YAML in the future.

However, Docker Compose v2.16 (compose-go v1.12.0) did support legacy YAML. If you are going to make breaking changes, you should do it when we bump the major version using semantic versioning.

As @ndeloof initially proposed, it is a good idea to display warnings in the current major version and make updates that will stop support in the future.
This approach will help users prepare for the changes and avoid any sudden disruptions.